### ISSUE TITLE
How to retrieve NPU per-layer/operator profiling from openvino_genai.LLMPipeline?

### ISSUE THREAD

**Issue by Vini-Pedroso** [openvinotoolkit/openvino.genai#4267]
# Summary

I am trying to collect per-layer/operator profiling for a decoder-only LLM running on Intel NPU through `openvino_genai.LLMPipeline`.

The model runs successfully on NPU, and `PERF_COUNT=YES` is accepted by the pipeline config. However, setting `NPU_PRINT_PROFILING=JSON` and `NPU_PROFILING_OUTPUT_FILE=...` does not create a profiling JSON file.

Using the model XML directly with `benchmark_app -pc` or `openvino.Core().compile_model(..., "NPU", {"PERF_COUNT": "YES"})` fails because the raw XML still contains dynamic shapes. So the GenAI pipeline appears to apply NPU-specific/static-shape preparation internally, but I cannot find a supported way to retrieve the underlying per-layer/operator profiling information from Python.

## Environment

- OS: Windows
- Python: 3.12
- OpenVINO: 2026.4.0-22520-ab6db2eed2a
- Device: Intel NPU
- Model: `OpenVINO/Mistral-7B-Instruct-v0.3-int4-cw-ov`
- Model path: `C:\models\mistral-7b-instruct-v03-int4-cw-ov`

## What works

The model runs successfully on NPU with `openvino_genai.LLMPipeline`.

Command:

    $env:NPU_PRINT_PROFILING="JSON"
    $env:NPU_PROFILING_OUTPUT_FILE="C:\Users\aipc-intel\Desktop\NAIA-MOAT\results_suite\mistral_npu_genai_profile.json"
    $env:NPU_PROFILING_VERBOSITY="HIGH"

    python scripts\run_openvino_genai_probe.py `
      -m C:\models\mistral-7b-instruct-v03-int4-cw-ov `
      -d NPU `
      -p "Diga oi em portugues." `
      -n 16 `
      --cache-dir C:\models\mistral-7b-instruct-v03-int4-cw-ov\.cache_NPU `
      --extra-config PERF_COUNT=YES

Output:

    OpenVINO devices: ['CPU', 'GPU', 'NPU']
    Model: C:\models\mistral-7b-instruct-v03-int4-cw-ov
    Device: NPU
    Pipeline config: {
      'MAX_PROMPT_LEN': 512,
      'MIN_RESPONSE_LEN': 32,
      'GENERATE_HINT': 'FAST_COMPILE',
      'CACHE_DIR': 'C:\\models\\mistral-7b-instruct-v03-int4-cw-ov\\.cache_NPU',
      'PERF_COUNT': True
    }
    Loading pipeline...
    Loaded in 49.795s
    Generating...

    === OUTPUT ===
    Oi! (pronounced oh-ee) is a common informal

    === TIMING ===
    device=NPU
    load_s=49.795
    generation_s=1.866
    requested_new_tokens=16
    approx_requested_tokens_per_s=8.57

But the profiling JSON is not created.

Command:

    Test-Path C:\Users\aipc-intel\Desktop\NAIA-MOAT\results_suite\mistral_npu_genai_profile.json

Result:

    False

## GenAI metrics available, but not per-layer/operator

If I call `generate()` with a list input, the returned `DecodedResults` exposes `perf_metrics`, such as TTFT, TPOT, IPOT, throughput and inference duration.

Example values:

    {
      "num_input_tokens": 12,
      "num_generated_tokens": 4,
      "load_time_ms": 48782.0,
      "ttft_ms": 853.06,
      "tpot_ms": 117.54,
      "ipot_ms": 293.41,
      "throughput_tokens_per_s": 8.50,
      "generate_duration_ms": 1206.61,
      "inference_duration_ms": 1173.64
    }

These are useful run-level metrics, but they do not provide per-layer/operator profiling.

## What fails

Trying to profile the raw model XML directly with `benchmark_app -pc` fails on NPU.

Command:

    benchmark_app `
      -m C:\models\mistral-7b-instruct-v03-int4-cw-ov\openvino_model.xml `
      -d NPU `
      -pc `
      -niter 1 `
      -nireq 1 `
      -shape "input_ids[1,1],attention_mask[1,1],position_ids[1,1],beam_idx[1]"

Output:

    Model inputs:
      input_ids      : i64 / [1,1]
      attention_mask : i64 / [1,1]
      position_ids   : i64 / [1,1]
      beam_idx       : i32 / [1]

    Model outputs:
      logits : f32 / [1,1,32768]

    Compilation failed. vclAllocatedExecutableCreate4 result: 0x78000004 - [NPU_VCL] Compiler returned msg:
    Exception from src\core\src\partial_shape.cpp:261:
    to_shape was called on a dynamic shape.

Using OpenVINO Runtime directly also fails similarly:

    compiled = core.compile_model(model, "NPU", {"PERF_COUNT": "YES"})

Error:

    Compilation failed. vclAllocatedExecutableCreate4 result: 0x78000004
    to_shape was called on a dynamic shape.

## Question

What is the supported way to retrieve per-layer/operator profiling for an LLM running on Intel NPU through `openvino_genai.LLMPipeline`?
Is there a Python or C++ API to access the underlying `InferRequest::get_profiling_info()` or NPU profiling output from `LLMPipeline`?
Should `NPU_PRINT_PROFILING=JSON` work with `LLMPipeline` when `PERF_COUNT=YES` is passed?
Is there a required additional config property for NPU profiling through GenAI?
Is direct `benchmark_app -pc` on the raw XML expected to fail for these NPU LLM models because GenAI performs additional static-shape/stateful preparation internally?
